File: desafio-02/leonardohenrique/python/desafio02.py
#função onde irá verificar se a lista de numeros determinados no argumento da função são primos
def numsprimos(n):
#cria uma lista e define como verdadeira
    primos = [True] * (n + 1)
#define os números 0 e 1 como falso já que não são primos
    primos[0] = primos[1] = False

#utiliza o algorítimo "Crivo de Eratóstenes" para verificar números não primos
    for i in range(2, int(n**0.5) + 1):
        if primos[i]:
            for j in range(i * i, n + 1, i):
                primos[j] = False
#chama a lista de numeros restantes e imprime todos numeros primos
    for num in range(n + 1):
        if primos[num]:
            print(num ,"é um numero primo!")

#inicia a função com o valor de 10000 números conforme o desafio
numsprimos(10000)

File: desafio-02/WRJ-sudo/c/desafio02.c
#include <stdio.h>

int main() {

    int num_primos[10001];

     num_primos[0] = 0;
     num_primos[1] = 0;
    //assume todos os numeros como primo
    for(int i = 2; i<=10000;i++){
        num_primos[i] = 1;

    }
    //verifica se os numeros a partir do 2 sao primos
    for (int i = 2; i <= 10000; i++) {
        int prox_numero;

        if(num_primos[i] == 1){

            for (prox_numero=i*i; prox_numero <= 10000; prox_numero++) {
                if(prox_numero%i==0){
                num_primos[prox_numero] = 0;


                }

            }
        }
    }
    //imprime todos os numeros primos
    for(int i = 0; i<=10000;i++){

        if(num_primos[i] == 1){
            printf("%d\n", i );
        }
    }

    return 0;


}
